### ISSUE TITLE
aliases not found when stored in .aliases files

### ISSUE THREAD

**Issue by michaelaye** [scopatz/xonsh#313]
I source in my aliases via an `.aliases` file, for having them both available in bash and zsh.
xonsh does not pick them up.
My `.bash_profile` part that sources them in looks like the usual bash idiom:
```bash
if [ -f ~/.aliases ]; then
    . ~/.aliases
```

**Comment by michaelaye** [comment]
for further evaluation of xonsh I would need a quick workaround to get my aliases in. is there one? I tried 'source .aliases' but that does not work.

**Comment by asmeurer** [comment]
I think you have to use `source-bash`.

**Comment by michaelaye** [comment]
still doesn't work:
```bash
klay6683@MacL2938 ~ $ source-bash .aliases
klay6683@MacL2938 ~ $ cdsrc
xonsh: subprocess mode: command not found: cdsrc
```

**Comment by michaelaye** [comment]
hm...
```bash
klay6683@MacL2938 ~ $ source-bash?
Traceback (most recent call last):
  File "/Users/klay6683/miniconda3/lib/python3.4/site-packages/xonsh/base_shell.py", line 42, in default
    self.execer.exec(code, mode='single', glbs=self.ctx)  # no locals
  File "/Users/klay6683/miniconda3/lib/python3.4/site-packages/xonsh/execer.py", line 127, in exec
    return exec(code, glbs, locs)
  File "<xonsh-code>", line 1, in <module>
NameError: name 'source' is not defined
```

**Comment by scopatz** [comment]
`source-bash` doesn't grab aliases.  (yes we probably need to fix this.)  You probably want to look at the `xonsh.aliases.bash_aliases()` function.  This could probably be generalize to accept a filename or a list of files. From there it would be super easy to make an alias command that does this. Or you could fix the `xonsh.aliases.source_bash()` function to also pull out aliases.  That might be even better.  Would you be willing to put in a PR to that effect?

**Comment by michaelaye** [comment]
Is it a declared functionality to parse and take on things from the bash config? Then this should be fixed in wherever you do the bash profile parsing. To be able to parse another file after launch of xonsh is sugar on top, I think. So which of above solutions is responsible for parsing the bash profile?

**Comment by michaelaye** [comment]
This works, gladly, to switch conda environments:
```python
source-bash activate py27
```

**Comment by scopatz** [comment]
Hi @michaelaye - we don't explicitly parse bashrc or bash profile.  Rather, we fire up Bash in interactive mode, print the environment and aliases, and then record the differences.  

* [`xonsh.environ.bash_env()`](https://github.com/scopatz/xonsh/blob/master/xonsh/environ.py#L475) does the environment part of this.
* [`xonsh.aliases.bash_aliases()`](https://github.com/scopatz/xonsh/blob/master/xonsh/aliases.py#L72) does the alias part of this.

Arguably this process should be streamlined.

**Comment by michaelaye** [comment]
okay, so why does a bash started in interactive mode not get my aliases then? Is .bash_profile not sourced then?

**Comment by scopatz** [comment]
Ahh we might need to do `bash -l` to get it to start up in login mode.

**Comment by scopatz** [comment]
Or maybe `bash -l -i` is the right thing.

**Comment by scopatz** [comment]
This should be a quick fix in these functions.

**Comment by scopatz** [comment]
I opened #319.  Would you mind trying this out?

**Comment by michaelaye** [comment]
yes, b/c I'm already doing that .;)

**Comment by scopatz** [comment]
This was closed by #319
